docs/DeveloperGuide.md

```diff
@@ -1,33 +1,244 @@
+`executeFilterCommand` will also check whether the user has entered a print limit using `FilterCommandSlicer`'s method 
+`getWordPrintLimitFromFilterCommand`. Subsequently, depending on the filter mode, `getTargetedWordTypes` or `getTargetedStringTags` will be called 
+and the returned array of strings will be passed to `WordsFilter`’s static methods `filterByType`, `filterByStartingString` and `filterByIncludedString`.
+Lastly, `filterCommandExecutor` will call the method `printFilterList`.
+
+The following sequence diagram shows how the components interact with each other for the scenario where the user issues 
+the command `filter -continue by\start limit\10 -cs -cg.`
+
+![UML Filter word sequence diagram](graphics/diagrams/Sequence_FilterWords.png)
+
+<p align = "center"><i><b>Figure 5: Interactions between components for the command filter -continue by\start limit\10 -cs -cg</b></i></p>
+   
+## Bunny class family
+![UML Bunny class diagram](graphics/diagrams/Class_diagram_bunny.png)
+<center><i>Figure 6:  Bunny ideas UML Class Diagram</i></center>
+The above class diagram describes the overall architecture of the bunny list functionalities. Recall that the term bunny refers to  plot ideas that have yet to be devloped. 
+The above classes provide the functionality of storing such ideas in an organised manner that can easily be searched, saved and loaded.
+
+The `BunnyList` class has the public ArrayList of bunnies `bunniesList` that is accessed by the `DeleteBunny` class method `deleteBunny` which removes 
+a selected bunny from the `bunniesList` ArrayList. Similarly, `bunniesList` is also accessed by the `BunnyFilter` class which 
+contains the `filterBunny` function which can filter through the list and obtain bunnies with specified keywords in the 
+idea or the genre using the command `filter bunny i\IDEA g\GENRE`, where the user may choose to omit either the `IDEA` 
+or the `GENRE` when running the command. 
+
+The `BunnySaver` class accesses the `bunniesList` and overwrites the current `bunny.txt` file in the data directory, 
+saving all `Bunny` objects into the file using the `saveAllBunny`  method. Bunny objects saved in that file can then 
+be read by the `BunnyLoader` class and added into the `bunniesList` ArrayList each time the program is started up, which is done 
+by calling the `loadBunnyFile` method.
+
+The `GenBunny` class can access the `bunniesList` as well. The function `pickRandomBunny` from the `GenBunny` class first randomly 
+generates an integer between 0 and the max number of `Bunny` idea in the `bunniesList` ArrayList. It then selects that indexed `Bunny` from the 
+`bunniesList` and returns it to the user. This allows the user to easily choose an idea to start working on without struggling to decide which idea to use.
+
+![UML BunnyList sequence diagram](graphics/diagrams/Sequence_diagram_bunny.png)
```

Comment:
![Fluffle1](https://user-images.githubusercontent.com/53225908/97660677-425d3080-1aad-11eb-93f6-e95e2ff90c96.JPG)
Just some minor details, I believe we need to take note if the dotted line touches the end of the activation bar. Also the solid line shouldn't cut into the activation bar.